apisix/plugins/ldap-auth.lua

```diff
@@ -139,8 +141,20 @@ function _M.rewrite(conf, ctx)
     local uid = conf.uid or "cn"
 
     local userdn =  uid .. "=" .. user.username .. "," .. conf.base_dn
-    local ld = lualdap.open_simple (conf.ldap_uri, userdn, user.password, conf.use_tls)
-    if not ld then
+    local ldapconf = {
+        timeout = 10000,
+        start_tls = false,
+        ldap_host = conf.ldap_host,
+        ldap_port = conf.ldap_port,
+        ldaps = conf.use_tls,
+        verify_ldap_host = conf.verify_ldap_host,
+        base_dn = conf.base_dn,
+        attribute = uid,
+        keepalive = 60000,
+    }
+    local res, err = ldap.ldap_authenticate(user.username, user.password, ldapconf)
+    if not res then
+        core.log.error(err)
```

Comment:
We use warning level log for this situation:
https://github.com/apache/apisix/blob/5cbcf246021b9f5a97ce98e999a7ed65a1cbc117/apisix/plugins/jwt-auth.lua#L369
and it would be better to add a prefix in the error message